src/b_setup.hpp

```diff
@@ -47,52 +49,52 @@ Mount mount(RA_STEPS_PER_DEGREE, DEC_STEPS_PER_DEGREE, &lcdMenu);
 WifiControl wifiControl(&mount, &lcdMenu);
 #endif
 
-void finishSetup();
-
 /////////////////////////////////
-//   ESP32
+//   Interrupt handling
 /////////////////////////////////
-#if defined(ESP32) && (RUN_STEPPERS_IN_MAIN_LOOP == 0)
-// Forward declare the two functions we run in the main loop
-void serialLoop();
+/* There are three possible configurations for periodically servicing the steper drives:
+ * 1) If RUN_STEPPERS_IN_MAIN_LOOP != 0 then the stepper drivers are called from Mount::loop().
+ *    Performance depends on how fast Mount::loop() can execute. With serial or UI activity it 
+ *    is likely that steps will be missed and tracking may not be smooth. No interrupts or threads
+ *    are used, so this is simple to get running.
+ * 2) If ESP32 is #defined then a periodic task is assigned to Core 0 to service the steppers.
+ *    stepperControlTask() is scheduled to run every 1 ms (1 kHz rate). On ESP32 the default Arduino 
+ *    loop() function runs on Core 1, therefore serial and UI activity also runs on Core 1. 
+ *    Note that Wifi and Bluetooth drivers will be sharing Core 0 with stepperControlTask().
+ *    This configuration decouples stepper servicing from other OAT activities by using both cores.
+ * 3) By default (e.g. for ATmega2560) a periodic timer is configured for a 500 us (2 kHz rate interval).
+ *    This timr generates interrupts which are handled by stepperControlCallback(). The stepper 
+ *    servicing therefore suspends loop() to generate motion, ensuring smooth tracking.
+ */
+#if (RUN_STEPPERS_IN_MAIN_LOOP != 0)
+  // Nothing to do - Mount::loop() will manage steppers in-line
+
+#elif defined(ESP32)
 
 TaskHandle_t StepperTask;
-TaskHandle_t  CommunicationsTask;
 
-/////////////////////////////////
-//
-// stepperControlFunc
-//
-// This task function is run on Core 1 of the ESP32
-/////////////////////////////////
+// This is the task for simulating periodic interrupts on ESP32 platforms. 
+// It should do very minimal work, only calling Mount::interruptLoop() to step the stepper motors as needed.
+// This task function is run on Core 0 of the ESP32 and never returns
 void IRAM_ATTR stepperControlTask(void* payload)
 {
   Mount* mount = reinterpret_cast<Mount*>(payload);
   for (;;) {
     mount->interruptLoop();
+    vTaskDelay(1);  // 1 ms 	// This will limit max stepping rate to 1 kHz
```

Comment:
Is there a specific reason you added this back in? We did some experimentation and found that removing this made the stepper run better....

Reply:
Just to add some context. we introduced this delay once because the esp32 watchdog caused a crash thinking that the core is never idling (i mean ... its true but its not an issue). But since we are using `disableCore0WDT()`, we don't have to delay here. The only thing to be still verified is if this somehow influences WiFi in a bad way since it is running on Core 0 per default iirc.

Reply:
I've added this back to allow some time on Core 0 for other tasks, like Wifi. stepperControlTask() is running at a slightly elevated priority, without this it will consume all time available on Core 0. In the FreeRTOS build for ESP32 1 ms is the scheduler period, so cannot go lower. I didn't see any issues with stepper smoothness - I will check again.

Reply:
i will have to investigate if this is still an issue ...